### ISSUE TITLE
REIXS crashes in MPlotPlotSelectorTool mouse press event

### ISSUE THREAD

**Issue by helfrij** [acquaman/acquaman#1975]
Not sure about what happened to cause the crash, some investigation required.

0	MPlotPlotSelectorTool::mousePressEvent	/home/reixs/beamline/programming/acquaman/../MPlot/lib/libMPlot.so.1	0	0x2acad862d871	
1	QGraphicsItem::sceneEvent	/home/beamline/tools/qt/qt-4.7.3-SL54-x86_64/lib/libQtGui.so.4	0	0x2acada02cb42	
2	??	/home/beamline/tools/qt/qt-4.7.3-SL54-x86_64/lib/libQtGui.so.4	0	0x2acada0504c8	
3	??	/home/beamline/tools/qt/qt-4.7.3-SL54-x86_64/lib/libQtGui.so.4	0	0x2acada050dad	
4	??	/home/beamline/tools/qt/qt-4.7.3-SL54-x86_64/lib/libQtGui.so.4	0	0x2acada0653db	
5	QGraphicsScene::mousePressEvent	/home/beamline/tools/qt/qt-4.7.3-SL54-x86_64/lib/libQtGui.so.4	0	0x2acada0655ec	
6	QGraphicsScene::event	/home/beamline/tools/qt/qt-4.7.3-SL54-x86_64/lib/libQtGui.so.4	0	0x2acada068c3a	
7	QApplicationPrivate::notify_helper	/home/beamline/tools/qt/qt-4.7.3-SL54-x86_64/lib/libQtGui.so.4	0	0x2acad9a0185f	
8	QApplication::notify	/home/beamline/tools/qt/qt-4.7.3-SL54-x86_64/lib/libQtGui.so.4	0	0x2acad9a08e32	
9	QCoreApplication::notifyInternal	/home/beamline/tools/qt/qt-4.7.3-SL54-x86_64/lib/libQtCore.so.4	0	0x2acadaa0ab64	
10	??	/home/beamline/tools/qt/qt-4.7.3-SL54-x86_64/lib/libQtGui.so.4	0	0x2acad9a01e0c	
11	QGraphicsView::mousePressEvent	/home/beamline/tools/qt/qt-4.7.3-SL54-x86_64/lib/libQtGui.so.4	0	0x2acada080a74	
12	QWidget::event	/home/beamline/tools/qt/qt-4.7.3-SL54-x86_64/lib/libQtGui.so.4	0	0x2acad9a5d50f	
13	QFrame::event	/home/beamline/tools/qt/qt-4.7.3-SL54-x86_64/lib/libQtGui.so.4	0	0x2acad9e3418b	
14	QAbstractScrollArea::viewportEvent	/home/beamline/tools/qt/qt-4.7.3-SL54-x86_64/lib/libQtGui.so.4	0	0x2acad9ec3657	
15	QGraphicsView::viewportEvent	/home/beamline/tools/qt/qt-4.7.3-SL54-x86_64/lib/libQtGui.so.4	0	0x2acada081481	
16	??	/home/beamline/tools/qt/qt-4.7.3-SL54-x86_64/lib/libQtGui.so.4	0	0x2acad9ec5798	
17	QCoreApplicationPrivate::sendThroughObjectEventFilters	/home/beamline/tools/qt/qt-4.7.3-SL54-x86_64/lib/libQtCore.so.4	0	0x2acadaa0aeff	
18	QApplicationPrivate::notify_helper	/home/beamline/tools/qt/qt-4.7.3-SL54-x86_64/lib/libQtGui.so.4	0	0x2acad9a01830	
19	QApplication::notify	/home/beamline/tools/qt/qt-4.7.3-SL54-x86_64/lib/libQtGui.so.4	0	0x2acad9a08fe7	
20	QCoreApplication::notifyInternal	/home/beamline/tools/qt/qt-4.7.3-SL54-x86_64/lib/libQtCore.so.4	0	0x2acadaa0ab64	
...	<More>

**Comment by helfrij** [comment]
Inspecting MPlotPlotSelectorTool::mousePressEvent(...), it isn't immediately obvious to me exactly what's causing the crash. I see one potential red flag where there's an assumption that all pointers returned by plot()->plotItems() are valid, problematic if there were any invalid items added to the plot. @dretrex, is this something that could happen?

**Comment by dretrex** [comment]
That is weird and it does appear that way doesn't it?  It might be worth looking into how plot items are added and removed and see if it is possible to have garbage pointers.

**Comment by helfrij** [comment]
Looking at MPlot::insertItem(...), it looks like MPlot would crash if an invalid item was added to a plot. If the problem is invalid items, the item would be a problem long before the plot selector tool comes into play.

**Comment by helfrij** [comment]
It's possible that a plot item could be added to the plot and deleted by someone else before the item is removed?

**Comment by dretrex** [comment]
Yes.  Just because the pointer is reparented by MPlot doesn't preclude the fact that others can have those pointers too.

**Comment by mButlerDM** [comment]
I've run into the same issue on every beam line when looking at a generic scan that has plots for more than 3 detectors. Selecting several different plots at random seems to cause a crash in an isolated, local environment.
